frame/staking/src/lib.rs

```diff
@@ -761,6 +766,105 @@ impl<AccountId, Balance: HasCompact + Zero> UnappliedSlash<AccountId, Balance> {
 	}
 }
 
+/// Something that defines the maximum number of nominations per nominator based on a curve.
```

Comment:
In a follow-up, you can also re-pot these types and traits into a new file, or here.